openpype/hosts/maya/plugins/publish/validate_resolution.py

```diff
@@ -0,0 +1,54 @@
+import pyblish.api
+from openpype.pipeline import (
+    PublishValidationError,
+    OptionalPyblishPluginMixin
+)
+from maya import cmds
+from openpype.hosts.maya.api.lib import reset_scene_resolution
+
+
+class ValidateResolution(pyblish.api.InstancePlugin,
+                         OptionalPyblishPluginMixin):
+    """Validate the resolution setting aligned with DB"""
+
+    order = pyblish.api.ValidatorOrder - 0.01
+    families = ["renderlayer"]
+    hosts = ["maya"]
+    label = "Validate Resolution"
+    optional = True
+
+    def process(self, instance):
+        if not self.is_active(instance.data):
+            return
+        width, height = self.get_db_resolution(instance)
+        current_width = cmds.getAttr("defaultResolution.width")
+        current_height = cmds.getAttr("defaultResolution.height")
+        if current_width != width and current_height != height:
+            raise PublishValidationError("Resolution Setting "
+                                         "not matching resolution "
+                                         "set on asset or shot.")
+        if current_width != width:
+            raise PublishValidationError("Width in Resolution Setting "
+                                         "not matching resolution set "
+                                         "on asset or shot.")
+
+        if current_height != height:
+            raise PublishValidationError("Height in Resolution Setting "
+                                         "not matching resolution set "
+                                         "on asset or shot.")
```

Comment:
Each of these reports should be showing the current value **and** the expected value so that the user knows what the differences are!

I also don't see a reason to really create a fully different report for which attribute mismatches. I feel like we can just log:

> Render resolution is 1920x1080 does not match asset resolution is 2100x1900